packages/template-retail-react-app/app/components/display-price/index.jsx

```diff
@@ -7,61 +7,193 @@
+
+    const renderPriceSet = (isRange) => (
+        <>
+            {renderCurrentPrice(isRange)} {isOnSale && renderListPrice(isRange)}
+        </>
+    )
+
+    if (isASet) {
+        return renderCurrentPrice(true)
+    }
+
+    if (isMaster) {
+        return renderPriceSet(isRange)
+    }
+
+    return <Box>{renderPriceSet(false)}</Box>
+}
+/**
+ * @private
+ * Component that displays current price of a product with a11y
+ * @param price - price of the product
+ * @param as - an HTML tag or component to be rendered as
+ * @param isRange - show price as range or not
+ * @param currency - currency to show the price in
+ * @param extraProps - extra props to be passed into Text Component
+ * @returns {JSX.Element}
+ */
+const CurrentPrice = ({price, as, isRange = false, currency, ...extraProps}) => {
     const intl = useIntl()
-    const {currency: activeCurrency} = useCurrency()
-    return (
-        <Skeleton isLoaded={basePrice} display={'flex'} {...skeletonProps}>
-            <Text fontWeight="bold" fontSize="md" mr={1}>
-                {isProductASet &&
-                    `${intl.formatMessage({
-                        id: 'product_view.label.starting_at_price',
-                        defaultMessage: 'Starting at'
-                    })} `}
-            </Text>
-            {typeof discountPrice === 'number' && (
-                <Text as="b" {...discountPriceProps}>
-                    {intl.formatNumber(discountPrice, {
-                        style: 'currency',
-                        currency: currency || activeCurrency
-                    })}
-                </Text>
-            )}
-            <Text
-                as={typeof discountPrice === 'number' ? 's' : 'b'}
-                ml={typeof discountPrice === 'number' ? 2 : 0}
-                fontWeight={discountPrice ? 'normal' : 'bold'}
-                {...basePriceProps}
-            >
-                {intl.formatNumber(basePrice, {
-                    style: 'currency',
-                    currency: currency || activeCurrency
-                })}
-            </Text>
-        </Skeleton>
+    const currentPriceText = intl.formatNumber(price, {
+        style: 'currency',
+        currency
+    })
+    return isRange ? (
+        <Text
+            as={as}
+            {...extraProps}
+            aria-label={intl.formatMessage(msg.ariaLabelCurrentPriceWithRange, {
+                currentPrice: currentPriceText
+            })}
+        >
+            {intl.formatMessage(msg.currentPriceWithRange, {
+                currentPrice: currentPriceText
+            })}
+        </Text>
+    ) : (
+        <Text
+            as={as}
+            aria-label={intl.formatMessage(msg.ariaLabelCurrentPrice, {
+                currentPrice: currentPriceText
+            })}
+        >
+            {intl.formatMessage(msg.currentPrice, {
+                currentPrice: currentPriceText
+            })}
+        </Text>
     )
```

Comment:
Another way to write this (which I think shows more clearly what differs and what stays the same).

```suggestion
    const ariaLabel = intl.formatMessage(
        isRange ? msg.ariaLabelCurrentPriceWithRange : msg.ariaLabelCurrentPrice,
        {
            currentPrice: currentPriceText
        }
    )
    const message = intl.formatMessage(isRange ? msg.currentPriceWithRange : msg.currentPrice, {
        currentPrice: currentPriceText
    })

    return (
        <Text as={as} {...extraProps} aria-label={ariaLabel}>
            {message}
        </Text>
    )
```